src/services/raspberrydevconn/app.py

```diff
@@ -0,0 +1,135 @@
+import io
+import logging
+import cherrypy
+import paho.mqtt.client as mqtt
+
+from common.WIOTRestApp import *
+from common.SettingsManager import *
+from common.SettingsNode import *
+from common.RESTBase import RESTBase
+from common.CatalogRequest import *
+from common.WIOThread import WIOThread
+
+class RaspberryDevConnAPI(RESTBase):
+
+    def __init__(self, upperRESTSrvcApp, settings: SettingsNode) -> None:
+        super().__init__(upperRESTSrvcApp, 0)
+        self._catreq = CatalogRequest(self.logger, settings)
+        self._th = WIOThread(target=self._airhumidity, name="Air Humidity Thread")
+        self._th1 = WIOThread(target=self._airtemperature, name="Air Temperature Thread")
+        self._th2 = WIOThread(target=self._terrainhumidity, name="Terrain Humidity Thread")
+        upperRESTSrvcApp.subsribe_evt_stop(self._th.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th1.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th2.stop)
+        f = open("../deviceconfig/confs.json")
+        data = json.load(f)
+        self.wait_temp_hum = data["sensors"]["temp"]["sampleperiod"] / 1000
+        self.logger.debug(self.wait_temp_hum)
+        self.wait_air_hum = data["sensors"]["airhum"]["sampleperiod"] / 1000
+        self.logger.debug(self.wait_temp_hum)
+        self.wait_soil_hum = data["sensors"]["soilhum"]["sampleperiod"] / 1000
+        self.logger.debug(self.wait_temp_hum)
+        self._th.run()
+        self._th1.run()
+        self._th2.run()
+        with open(SettingsManager.relfile2abs("../deviceconfig/confs.json")) as fp:
+                keys = self._rec_dict(json.load(fp))
+                for k in keys:
+                    self._catreq.subscribeMQTT("DeviceConfig", f"/conf{k}")
+                    if (f"/conf{k}" == "/conf/sensors/temp/sampleperiod"):
+                        self._catreq.callbackOnTopic("DeviceConfig", f"/conf{k}", self.onMessageReceiveTemp)
+                    elif (f"/conf{k}" == "/conf/sensors/airhum/sampleperiod"):
+                        self._catreq.callbackOnTopic("DeviceConfig", f"/conf{k}", self.onMessageReceiveAirhum)
+                    elif (f"/conf{k}" == "/conf/sensors/soilhum/sampleperiod"):
+                        self._catreq.callbackOnTopic("DeviceConfig", f"/conf{k}", self.onMessageReceiveSoilhum)
+        
+    def _airhumidity(self):
+        while not self._th.is_stop_requested:
+            self._th.wait(self.wait_air_hum)                                                   
+            self._catreq.publishMQTT("RaspberryDevConn", "/airhumidity", "airhumiditypayload")
+    
+    def _airtemperature(self):
+        while not self._th.is_stop_requested:
+            self._th.wait(self.wait_temp_hum)                                                   
```

Comment:
Here you should use the right WIOThread object so for example:

```suggestion
            self._th1.wait(self.wait_temp_hum)                                                   
```

Reply:
Yes you are right.

Reply:
@giop98 there is the same problem in the third thread too, I think

Reply:
@thegabriele97 Yes, today I will go over and check all threads

Reply:
Fix in 91f92ba702af91a1c813d49d46d8caeaa907b9b1